Kudu.Services/ServiceHookHandlers/OneDriveHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Web;
+using Kudu.Contracts.Settings;
+using Kudu.Contracts.Tracing;
+using Kudu.Core;
+using Kudu.Core.Deployment;
+using Kudu.Core.SourceControl;
+using Kudu.Services.FetchHelpers;
+using Newtonsoft.Json.Linq;
+
+namespace Kudu.Services.ServiceHookHandlers
+{
+    public class OneDriveHandler : IServiceHookHandler
+    {
+        private readonly OneDriveHelper _oneDriveHelper;
+
+        public OneDriveHandler(ITracer tracer,
+                              IDeploymentSettingsManager settings,
+                              IEnvironment environment)
+        {
+            _oneDriveHelper = new OneDriveHelper(tracer, settings, environment);
+        }
+
+        public DeployAction TryParseDeploymentInfo(HttpRequestBase request, JObject payload, string targetBranch, out DeploymentInfo deploymentInfo)
+        {
+            deploymentInfo = null;
+            string url = payload.Value<string>("url");
+            if (string.IsNullOrWhiteSpace(url) || !url.ToLowerInvariant().Contains("api.onedrive.com"))
+            {
+                return DeployAction.UnknownPayload;
+            }
+
+            if (string.IsNullOrWhiteSpace(payload.Value<string>("access_token")))
+            {
+                return DeployAction.NoOp;
+            }
+
```

Comment:
Let this to be null ref.   Our contract is strict and we want unexpected thing to fail miserably.   See github for instance.